server/game/cards/13.5-TB/WhiteCloak.js

```diff
@@ -0,0 +1,32 @@
+const DrawCard = require('../../drawcard.js');
+
+class WhiteCloak extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction({ controller: 'current', trait: ['Knight'] });
+        this.whileAttached({
+            effect: [
+                ability.effects.addTrait('Kingsguard')
+            ]
+        });
+        this.interrupt({
+            canCancel: true,
+            when: {
+                onCharacterKilled: event => event.allowSave && event.card.canBeSaved() && this.isKingOrQueen(event.card)
+            },
+            cost: ability.costs.kneelParent(),
+            handler: context => {
+                let parent = context.cardStateWhenInitiated.parent;

```

Comment:
Just FYI, this probably wasn't strictly necessary. It's only important when the parent character is killed or sacrificed as a cost (e.g. Glamor), and thus becomes unattached before the effect of the ability resolves. But this should still work.